test/Datadog.Trace.Tests/Logging/SerilogLogProviderTests.cs

```diff
@@ -103,6 +104,56 @@ public void DisabledLibLogSubscriberDoesNotAddCorrelationIdentifiers()
             Assert.All(_logEvents, e => LogEventDoesNotContainCorrelationIdentifiers(e));
         }
 
+        [Fact]
+        public void LogTheRightSpan()
+        {
+            // Assert that the Serilog log provider is correctly being used
+            Assert.IsType<CustomSerilogLogProvider>(LogProvider.CurrentLogProvider);
+
+            // Instantiate a tracer for this test with default settings and set LogsInjectionEnabled to TRUE
+            var tracer = LoggingProviderTestHelpers.InitializeTracer(enableLogsInjection: true);
+
+            var barrier1 = new Barrier(2);
+            var barrier2 = new Barrier(2);
+
+            var task1 = Task.Run(() => Thread1(barrier1, tracer));
+            barrier1.SignalAndWait(); // Wait until thread1 has created both scopes
+
+            var task2 = Task.Run(() => Thread2(barrier2, tracer));
+            barrier2.SignalAndWait(); // Wait until thread2 has created both scopes
+
+            // Stack is: Empty -> outer1 -> Empty -> outer2
```

Comment:
How did the second Empty get here?

Reply:
Also, instead of committing comments that document the problematic state, can we document the expected state?

Reply:
Serilog context is asynclocal, so when the first span is pushed for the second thread, the context is still empty.